src/engine/serverbrowser.cpp

```diff
@@ -309,12 +292,9 @@ void checkresolver()
     int resolving = 0;
     loopv(servers)
     {
-        serverinfo &si = *servers[i];
-        if(si.resolved == serverinfo::RESOLVED) continue;
-        if(si.address.host == ENET_HOST_ANY)
+        if( servers[i]->need_resolve( resolving ) )
```

Comment:
From the method name and its param, I could not guess what exactly that method does. You could get away by incrementing the `resolving` variable inside this method. That's what was done before, too. I checked the implementation, and I consider it bad practice the way it's done now.